spring-vault-core/src/main/java/org/springframework/vault/core/lease/SecretLeaseContainer.java

```diff
@@ -463,10 +474,15 @@ protected Lease doRenewLease(RequestedSecret requestedSecret, final Lease lease)
 						@Override
 						@SuppressWarnings("unchecked")
 						public ResponseEntity<Map<String, Object>> doWithRestOperations(
-								RestOperations restOperations) {
+										RestOperations restOperations) {
```

Comment:
It could make rather sense to use `onLeaseExpired` instead of adding conditions to handle generic secrets. `onLeaseExpired` will handle rotation (request new secrets) already. 

`LeaseAwareVaultPropertySource` clears right now the contents of the property source on expiry to prevent stale data.